api/src/services/agencies_v1/get_agencies.py

```diff
@@ -26,15 +29,28 @@ class AgencyListParams(BaseModel):
 
 
 def get_agencies(
-    db_session: db.Session, list_params: AgencyListParams
+    db_session: db.Session, list_params: AgencyListParams, active: bool | None = None
 ) -> Tuple[Sequence[Agency], PaginationInfo]:
 
+    import pdb; pdb.set_trace()
+
     stmt = (
-        select(Agency).options(joinedload(Agency.top_level_agency), joinedload("*"))
-        # Exclude test agencies
-        .where(Agency.is_test_agency.isnot(True))
+    select(Agency).options(joinedload(Agency.top_level_agency), joinedload("*"))
+    # Exclude test agencies
+    .where(Agency.is_test_agency.isnot(True))
     )
 
+    if active:
+        stmt = (
+            select(Agency)
+            .join(Opportunity, onclause=Agency.agency_code==Opportunity.agency_code)
+            .join(CurrentOpportunitySummary)
+            .join(LkOpportunityStatus)
+            .where(Agency.is_test_agency.isnot(True)) # Exclude test agencies
+            .where(LkOpportunityStatus.opportunity_status_id.in_([1,2])) # Opportunities associated with posted or forecasted status
```

Comment:
I assume this is still a work in progress, but you can make this a bit simpler of a query / use the enums directly:
```suggestion
            .join(CurrentOpportunitySummary)
            .where(Agency.is_test_agency.isnot(True)) # Exclude test agencies
            .where(CurrentOpportunitySummary.opportunity_status.in_([OpportunityStatus.FORECASTED,OpportunityStatus.POSTED])) # Opportunities associated with posted or forecasted status
```

Reply:
Yup, draft query, updated!